### PR TITLE
Port jlm to LLVM 15

### PR DESCRIPTION
**Pull request by phate**
Main changes:
1. PointerType became opaque. This triggered an avalanche of changes
2. Minor changes for alignment
3. Attributes changed

### CODE REVIEW

**Comment by phate** [comment]
@sjalander : What do I need to do to get hls-test-suite working again?

File: libjlm/src/backend/llvm/rvsdg2jlm/rvsdg2jlm.cpp [hunk 1]
```diff
@@ -496,7 +499,7 @@ convert_delta_node(const jive::node & node, context & ctx)
 	auto dnode = data_node::Create(
 		m.ipgraph(),
 		delta->name(),
-		delta->type(),
+    delta->type(),
```
Comment (sjalander) [inline libjlm/src/backend/llvm/rvsdg2jlm/rvsdg2jlm.cpp:502]: This change looks like noise

**Review by sjalander — APPROVED** [review]

**Comment by sjalander** [comment]
Do you also add the change you made yesterday:
diff --git a/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp b/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp
index b29da4b8..36404ba6 100644
--- a/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp
+++ b/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp
@@ -243,7 +243,7 @@ jlm::hls::split_hls_function(jlm::RvsdgModule &rm, const std::string &function_n
                     smap.insert(ln->input(i)->origin(), arg);
                     // add export for delta to rm
                     // TODO: check if not already exported and maybe adjust linkage?
-                    rm.Rvsdg().add_export(odn->output(), {odn->type(), odn->name()});
+                    rm.Rvsdg().add_export(odn->output(), {odn->output()->type(), odn->name()});
                 } else {
                     throw jlm::error("Unsupported node type: " + orig_node->operation().debug_string());
                 }

File: libjlm/src/backend/llvm/rvsdg2jlm/rvsdg2jlm.cpp (hunk 1 above)
Comment (phate) [reply libjlm/src/backend/llvm/rvsdg2jlm/rvsdg2jlm.cpp:502]: Done.

**Comment by phate** [comment]
> Do you also add the change you made yesterday: diff --git a/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp b/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp index b29da4b8..36404ba6 100644 --- a/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp +++ b/libjlm/src/backend/hls/rvsdg2rhls/rvsdg2rhls.cpp @@ -243,7 +243,7 @@ jlm::hls::split_hls_function(jlm::RvsdgModule &rm, const std::string &function_n smap.insert(ln->input(i)->origin(), arg); // add export for delta to rm // TODO: check if not already exported and maybe adjust linkage?
> 
>     * ```
>                      rm.Rvsdg().add_export(odn->output(), {odn->type(), odn->name()});
>       ```
> 
> 
>     * ```
>                      rm.Rvsdg().add_export(odn->output(), {odn->output()->type(), odn->name()});
>                  } else {
>                      throw jlm::error("Unsupported node type: " + orig_node->operation().debug_string());
>                  }
>       ```

Added.

**Review by sjalander — APPROVED** [review]

**Pull request merged**